netdev: fix concurrency issues with multiple sockets

scottfeldman opened the pull request:
The wifinina driver was not handling multiple concurrent sockets correctly.  This PR fixes the wifinina driver and adds an example/net/snake test for testing multiple concurrent sockets.  The wioterminal driver (rtl8270dn) needed no changes to pass the snake test.

scottfeldman committed:
add classic network snake test example

Adding this example to test multiple concurrent socket connections.

deadprogram commented on examples/net/snake/main.go:
```diff
@@ -0,0 +1,133 @@
+	server string = "10.0.0.100:8080"
+)
+
+// Boa Constrictor, a poem by Shel Silverstein
```
As much as I enjoy the poem, I do not think we have usage rights to include this.

deadprogram commented:
I tested these changes and they appear to be working as expected.

scottfeldman committed:
replace copyrighted poem with AI generated one

scottfeldman committed:
remove poem and just use main.go as text source

Even AI-generated text may be copyrighted in some indirect way, so avoid
the whole copyright issue and just feed main.go to the snake.

deadprogram commented:
Rebased to squash the last 3 commits together, then merged to `dev`. Thanks for the fixes @scottfeldman
